crm_request/models/request.py

```diff
@@ -103,10 +103,21 @@ def message_new(self, msg, custom_values=None):
         alias = self.env['mail.alias'].search(
             [['alias_name', '=', alias_char]])
 
+        # Find the corresponding type
+        subject = msg.get('subject')
+        type_ids = self.env['crm.claim.type'].search(
+            [('keywords', '!=', False)])
+        type_id = False
+        for record in type_ids:
+            if any(word in subject for word in record.get_keys()):
+                type_id = record.id
+                break
```

Comment:
Would it be worth it to make a statistical analysis of the keywords ? For exemple: There was 3 word in the subject that matched type1 and 1 word that matched type 2 so the type that is set is type1 ...